org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/cvc_complex_type_2_4_bCodeAction.java

```diff
@@ -0,0 +1,138 @@
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+import org.eclipse.lsp4j.Position;
+import org.w3c.dom.NodeList;
+
+/**
+ * CodeAction to insert expected child elements within an element
+ * 
+ * Given this XML where the expected child elements are not present as defined
+ * in the XSD:
+ * 
+ * <servlet></servlet>
+ * Error:
+ * Child elements are missing from element:
+ * - servlet
+ *
+ * The following elements are expected:
+ * - description
+ * - display-name
+ * - icon
+ * - servlet-name
+ * 
+ * To fix the error, the code action will suggest inserting the expected
+ * elements inside the parent tag
+ * 
+ */
+
+public class cvc_complex_type_2_4_bCodeAction implements ICodeActionParticipant {
+
+	@Override
+	public void doCodeAction(ICodeActionRequest request, List<CodeAction> codeActions, CancelChecker cancelChecker) {
+		DOMDocument document = request.getDocument();
+		Diagnostic diagnostic = request.getDiagnostic();
+		try {
+			Range diagnosticRange = diagnostic.getRange();
+			int startOffset = document.offsetAt(diagnosticRange.getStart()) + 1;
+			DOMNode node = document.findNodeAt(startOffset);
+
+			if (node == null || !node.isElement()) {
+				return;
+			}
+
+			DOMElement element = (DOMElement) node;
+			int elementOffset = element.getEndTagOpenOffset();
+			Position childElemenPosition = document.positionAt(elementOffset);
+
+			String insertText = "";
+
+			XMLGenerator generator = request.getXMLGenerator();
+			ContentModelManager contentModelManager = request.getComponent(ContentModelManager.class);
+
+			for (CMDocument cmDocument : contentModelManager.findCMDocument(element)) {
+				Collection<CMElementDeclaration> cmElements = cmDocument.getElements();
+
+				Collection<CMElementDeclaration> childElements;
+				try {
+					childElements = getChildElements(cmElements, generator);
+					for (CMElementDeclaration expectedElement : childElements) {
+						insertText = insertText.concat(generator.generate(expectedElement, null, 1, true));
+					}
+				} catch (Exception e) {
+					// do nothing
+				}
+
+			}
+
+			CodeAction insertExpectedElement = CodeActionFactory.insert(
+					"Insert all expected elements", childElemenPosition, insertText,
+					document.getTextDocument(), diagnostic);
+
+			codeActions.add(insertExpectedElement);
+
+		} catch (
+
+		BadLocationException e) {
+			// do nothing
+		}
+
+	}
+
+	/**
+	 * Returns the possible child elements for the given DOM element.
+	 * 
+	 */
+	private static Collection<CMElementDeclaration> getChildElements(Collection<CMElementDeclaration> cmElements,
+			XMLGenerator generator)
+			throws Exception {
+		List<CMElementDeclaration> possibleElements = new ArrayList<>();
+
+		for (CMElementDeclaration cmElement : cmElements) {
+			Collection<CMElementDeclaration> childElements = cmElement.getElements();
+			if (childElements.size() != 0) {
+				possibleElements.addAll(childElements);
```

Comment:
I initially created this function to recursively get the element declaration of all the elements and its children, but at the moment it doesn't work as I wanted and just returns the top level elements. It seems like generate() takes care of the children, so I'm not sure if I need to keep trying to get this function to work recursively.